### PR TITLE
Make code compatible with Python 2.7 and Python >=3.6

### PR DESCRIPTION
**Pull request by ilyeSudo**
1- Run modernize command to turn py2 code into a six-compatible code (py27 + py36)
2- Edit result manually
3- Test code and dependencies installation with `tox` command for both python versions

### CODE REVIEW

**Comment by ilyeSudo** [comment]
Thanks for the remark @charles-vdulac, so for CircleCI, it automatically runs test commands when   `tox.ini` is detected, thus I added it, please check it out.

**Comment by ilyeSudo** [comment]
After adding `tox.ini` file, Tox is having troubles with installing deps in py36 env.
I'm digging into it...

**Comment by rcshubhadeep** [comment]
> so for CircleCI, it automatically runs test commands when tox.ini is detected

I am not sure I understood this part. I checked the Circle CI build logs and it does not seem like that (apart from installing Tox) it did anything related to running the `tox` command. It did run the `nosetest` command as expected. Am I missing something?

**Comment by ilyeSudo** [comment]
So what I meant by that @rcshubhadeep is that CircleCI will run the tests in `[testenv]` section in `tox.ini` as explained in[ the official docs ](https://circleci.com/docs/1.0/language-python/#testing)

**Comment by xav-b** [comment]
@ilyeSudo it's indeed in the doc but looking at [circleci test steps](https://circleci.com/gh/Kpler/scrapy-datadog-extension/20) only `nosetests` was ran.

I wouldn't debug Circleci framework, just invoke `tox` from `circle`.

That being said the command in `tox` is very different from the command in `circle` and we loose coverage and reporting.

**Comment by rcshubhadeep** [comment]
As @xav-b already pointed out, even if it is in the docs it does not seem like they actually ran it. So maybe running it manually is better.

**Comment by ilyeSudo** [comment]
@xav-b the reason why `tox.ini` didn't run in the last build was because of the missing `tox` command in the `circle.yml` file.
I kept `nosetests` only for CircleCi to get use of the coverage and reporting as you mentioned, however, I set deps tests in `tox.ini` to keep things separate and I think we are good now! what do you think?

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+skipsdist = True
+
+[testenv]
+whitelist_externals = mkdir
+deps =
+  -rdev-requirements.txt
+  -rrequirements.txt
+
+commands=
+  mkdir -p {toxworkdir}/tests/{envname}/nosetests
```
Comment (rcshubhadeep) [inline tox.ini]: From `tox` documentation 

> toxworkdir=path   # tox working directory, defaults to {toxinidir}/.tox

If you check your build log, you will see that it indeed creates a directory at 

> py27 create: /home/ubuntu/scrapy-datadog-extension/.tox/py27

But it is not visible in the artifact tab in circle-ci. Maybe you need to set these paths properly.

File: dev-requirements.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ nose==1.3.7
 nose-cov==1.6
 nose-timer==0.7.1
 flake8==3.5.0
-
+tox==3.0.0
+tox-pyenv==1.1.0
```
Comment (rcshubhadeep) [inline dev-requirements.txt]: Not completely related, but can you please sort the entries in this file alphabetically?

**Pull request merged**